### PR TITLE
DEV014 setup devise authentication

### PR DESCRIPTION
**Pull request by MariaPaharelava**
DEV014 Setup devise authentication
[card](https://trello.com/c/pO9Yo0t7)

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -34,8 +34,8 @@
   end
 
   create_table "images", force: :cascade do |t|
-    t.integer "imageable_id", null: false
```
Comment (Maxxx911) [inline db/schema.rb]: Please, rolback this

Comment (MariaPaharelava) [reply db/schema.rb]: I didn't write this string

File: Gemfile [hunk 2]
```diff
@@ -20,7 +20,8 @@ gem 'active_model_serializers'
 gem 'thor', '~> 0.19.1'
 # See https://github.com/rails/execjs#readme for more supported runtimes
 # gem 'mini_racer', platforms: :ruby
-
+gem 'devise', '~> 4.2'
+gem 'pry', '~> 0.12.2'
```
Comment (andreydro) [inline Gemfile]: add ```gem 'pry-byebug'
  gem 'pry-rails'``` to development, test group

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -6,10 +6,11 @@
     <%= csp_meta_tag %>
 
     <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload' %>
-    <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
```
Comment (andreydro) [inline app/views/layouts/application.html.erb]: Why do you delete this line?

File: db/migrate/20190813134515_add_devise_to_users.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    add_column :users, :confirmed_at,:datetime
+    add_column :users, :confirmation_sent_at, :datetime
+    #add_column :users, :unconfirmed_email string# Only if using reconfirmable
+
+    ## Lockable
+    add_column :users, :failed_attempts, :integer, default: 0 # Only if lock strategy is :failed_attempts
+    add_column :users, :unlock_token, :string # Only if unlock strategy is :email or :both
+    add_column :users, :locked_at, :datetime
+
+    change_column :users, :nickname, :string, default: '', null: true
```
Comment (andreydro) [inline db/migrate/20190813134515_add_devise_to_users.rb]: Change 32-35 lines to `null: false`

Comment (andreydro) [reply db/migrate/20190813134515_add_devise_to_users.rb]: But remember at first rollback, than change, than migrate again

Comment (MariaPaharelava) [reply db/migrate/20190813134515_add_devise_to_users.rb]: Why?

File: Gemfile (hunk 2 above)
Comment (longbeardmen) [inline Gemfile]: move pry to `test` and `dev` groups

**Pull request merged**